specifyweb/frontend/js_src/lib/components/Forms/Save.tsx

```diff
@@ -225,7 +286,29 @@ export function SaveButton<SCHEMA extends AnySchema = AnySchema>({
             copyButton(
               formsText.carryForward(),
               formsText.carryForwardDescription(),
-              async () => resource.clone(false)
+              resource.specifyTable.name === 'CollectionObject' &&
+                carryForwardAmount > 2
+                ? async () => {
+                    const clones = Array.from(
+                      { length: carryForwardAmount },
+                      async () => {
+                        const clonedResource = await resource.clone(false);
+                        const formatter = clonedResource.specifyTable
+                          .strictGetLiteralField('catalogNumber')
+                          .getUiFormatter()!;
+                        const wildCard = formatter.valueOrWild();
```

Comment:
move this code out of the Array.from() since it will produce the same result for each iteration of the loop (because all resources are of the same table)

Reply:
I think this code is meant to avoid for loops when cloning a resource `carryForwardAmount` of times. What would be a better way to do it?

Reply:
to clarify, I mean that these two variables don't have to be re-computed on each iteration of the loop since they will be the same.
instead, define them above the lop, and use them inside the loop

```ts
                        const formatter = tables.CollectionObject
                          .strictGetLiteralField('catalogNumber')
                          .getUiFormatter()!;
                        const wildCard = formatter.valueOrWild();
```